server/src-lib/Hasura/Server/Init.hs

```diff
@@ -1084,23 +1085,13 @@ serveOptionsParser =
 -- and catalog schema versions.
 downgradeShortcuts :: [(String, String)]
 downgradeShortcuts = 
-  [ ("v1.0.0-beta.1", "16")
-  , ("v1.0.0-beta.2", "17")
-  , ("v1.0.0-beta.3", "17")
-  , ("v1.0.0-beta.4", "19")
-  , ("v1.0.0-beta.5", "19")
-  , ("v1.0.0-beta.6", "22")
-  , ("v1.0.0-beta.7", "24")
-  , ("v1.0.0-beta.8", "26")
-  , ("v1.0.0-beta.9", "26")
-  , ("v1.0.0-beta.10", "27")
-  , ("v1.0.0-rc.1", "28")
-  , ("v1.0.0", "28")
-  , ("v1.1.0-beta.1", "29")
-  , ("v1.1.0-beta.2", "30")
-  , ("v1.1.0-beta.3", "31")
-  , ("v1.1.0", "31")
-  ]
+  $(do addDependentFile "catalog_versions"
+       s <- runIO (readFile "catalog_versions")
+       let parseVersions = mapMaybe (parseVersion . words) . lines
+    
+           parseVersion [tag, version] = Just (tag, version)
+           parseVersion _ = Nothing
+       TH.lift (parseVersions s))     
```

Comment:
I’d rather this raise an error if parsing fails than silently drop the line, since the latter seems like it could cause headaches.

Reply:
Done